packages/zoe/test/unitTests/zcf/test-zcf.js

```diff
@@ -194,3 +188,202 @@ test(`zcf.getAmountMath - from issuerKeywordRecord & zcf.saveIssuer`, async t =>
   await zcf.saveIssuer(bucksKit.issuer, 'C');
   compareAmountMath(t, zcf.getAmountMath(bucksKit.brand), bucksKit.amountMath);
 });
+
+test(`zcf.assertUniqueKeyword`, async t => {
+  const { moolaKit, simoleanKit } = setup();
+  const issuerKeywordRecord = { A: moolaKit.issuer, B: simoleanKit.issuer };
+  const { zcf } = await setupZCFTest(issuerKeywordRecord);
+  t.throws(() => zcf.assertUniqueKeyword('A'), {
+    message: 'keyword (a string) must be unique\nSee console for error data.',
+  });
+  t.throws(() => zcf.assertUniqueKeyword('B'), {
+    message: 'keyword (a string) must be unique\nSee console for error data.',
+  });
+  // Unique, but not a valid Keyword
+  t.throws(() => zcf.assertUniqueKeyword('a'), {
+    message:
+      'keyword "a" must be ascii and must start with a capital letter.\nSee console for error data.',
+  });
+  t.throws(() => zcf.assertUniqueKeyword('3'), {
+    message:
+      'keyword "3" must be ascii and must start with a capital letter.\nSee console for error data.',
+  });
+});
+
+test(`zcf.saveIssuer & zoe.getTerms`, async t => {
+  const { moolaKit, simoleanKit, bucksKit } = setup();
+  const expected = {
+    brands: { A: moolaKit.brand, B: simoleanKit.brand, C: bucksKit.brand },
+    issuers: { A: moolaKit.issuer, B: simoleanKit.issuer, C: bucksKit.issuer },
+    maths: {
+      A: moolaKit.amountMath,
+      B: simoleanKit.amountMath,
+      C: bucksKit.amountMath,
+    },
+    whatever: 'whatever',
+  };
+  const issuerKeywordRecord = { A: moolaKit.issuer, B: simoleanKit.issuer };
+  const customTerms = {
+    whatever: 'whatever',
+  };
+
+  const { zcf, instance, zoe } = await setupZCFTest(
+    issuerKeywordRecord,
+    customTerms,
+  );
+  await zcf.saveIssuer(bucksKit.issuer, 'C');
+
+  const zoeTerms = await E(zoe).getTerms(instance);
+  const zoeTermsMinusAmountMath = { ...zoeTerms, maths: {} };
+  const expectedMinusAmountMath = { ...expected, maths: {} };
+  t.deepEqual(zoeTermsMinusAmountMath, expectedMinusAmountMath);
+
+  compareAmountMaths(t, zoeTerms.maths, expected.maths);
+});
+
+test(`zcf.saveIssuer - bad issuer`, async t => {
+  const { moolaKit } = setup();
+  const { zcf } = await setupZCFTest();
+  // @ts-ignore
+  await t.throwsAsync(() => zcf.saveIssuer(moolaKit.brand, 'A'), {
+    // TODO: improve error message
+    // https://github.com/Agoric/agoric-sdk/issues/1701
+    message: 'o["getBrand"] is not a function',
+  });
+});
+
+test(`zcf.saveIssuer - bad keyword`, async t => {
+  const { moolaKit } = setup();
+  const { zcf } = await setupZCFTest();
+  // TODO: why does this not throwAsync?
```

Comment:
The assertion in saveIssuer comes before the async portion, so it throws immediately.

Reply:
Ah, that seems like unpredictable behavior from a user's perspective. I'm going to change this to an async function such that it's always `throwsAsync` if it throws.